actions/archive2disk/v1/pkg/archive/archive.go

```diff
@@ -3,28 +3,30 @@ package archive
 // This package handles the pulling and management of images
 
 import (
-	"archive/tar"
-	"compress/gzip"
 	"fmt"
-	"io"
 	"net/http"
-	"os"
 	"path/filepath"
+	"time"
+	"strings"
 
 	log "github.com/sirupsen/logrus"
 )
 
 // Write will pull an image and write it to local storage device
-// with compress set to true it will use gzip compression to expand the data before
-// writing to an underlying device
-func Write(archiveURL, archiveType, path string) error {
+// image must be a tar file or tar.gz file as set by archiveType
+func Write(archiveURL, archiveType, path string, checksum string) error {
 
 	req, err := http.NewRequest("GET", archiveURL, nil)
 	if err != nil {
 		return err
 	}
 
-	resp, err := http.DefaultClient.Do(req)
+	//timeout added due to clients timing out when provisioning
+	//multiple systems at once.  May want to add this as an
```

Comment:
5 minutes seems like a reasonable default, but I do worry about adding a timeout where there is not one without a means to alter it.  Would you mind adding a variable for this in your PR? (Or opening a new issue to address this if not)

Reply:
Good call, I will add an environment variable here and push an update to the PR.

Reply:
`5 * time.Minute` reads better and doesn't require any math. I think it would be good to use the clearer value.

Reply:
I changed it to 

`client := &http.Client{Timeout: time.Second * time.Duration(httpTimeoutVal)}`

Where httpTimeoutVal is set from an environment variable with a default of 30.
I'm still doing some testing before pushing the change.